snuba/consumers/consumer.py

```diff
@@ -402,6 +507,13 @@ def close(self) -> None:
                 try:
                     step.close()
                 except Exception as e:
+                    messages = self.__messages[storage_key]
+                    if self.__dead_letter_step and messages:
```

Comment:
Good idea to track a metric of how many messages have been written to dead letter topic per storage.